### PR TITLE
TS-4845: NULL dereference in url_sig

### PR DESCRIPTION
**Pull request by jrushford**
Fix a NULL dereference in url_sig.

### CODE REVIEW

**Comment by jrushford** [comment]
[approve ci]

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/799/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/694/ for details.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/800/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/695/ for details.

File: plugins/experimental/url_sig/url_sig.c [hunk 1]
```diff
@@ -561,7 +561,9 @@ TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
 
 /* ********* Allow ********* */
 allow:
-  app_qry = getAppQueryString(query, strlen(query));
+  if (query != NULL) {
+    app_qry = getAppQueryString(query, strlen(query));
+  }
```
Comment (jpeach) [inline plugins/experimental/url_sig/url_sig.c]: So this is the case where the exclusion regex was matched so it is ok to not have a query string?

Comment (jrushford) [inline plugins/experimental/url_sig/url_sig.c]: Yes, thanks for the catch!

**Comment by jpeach** [comment]
👍

**Pull request merged**